backend/src/v4/models/unityAssets.js

```diff
@@ -76,4 +76,40 @@ UnityAssets.getUnityBundle = function(account, model, id) {
 	return FilesManager.getFileAsStream(account, collection, bundleFileName);
 };
 
+UnityAssets.getRepoBundle = function(account, model, id) {
+	const bundleFileName = `${id}`;
+	const collection = `${model}.stash.repobundles.ref`;
+	return FilesManager.getFileAsStream(account, collection, bundleFileName);
+};
+
+UnityAssets.getTexture = async function(account, model, id) {
+	const textureFilename = `${id}`;
+	const collection = `${model}.scene`;
+
+	const node = await db.findOne(account, collection, { _id: utils.stringToUUID(textureFilename) }, {
+		_id: 1,
+		_blobRef: 1,
+		extension: 1
+	});
+
+	const {elements, buffer} = node._blobRef;
```

Comment:
the server is going to crash if node is undefined here.

We need an if statement to make sure we found a matching node before we proceed, and respond accordingly if we failed to find a matching texture with that id.

Reply:
it will also do weird thigns if we found a matching node but it doesn't have a blobRef (e.g. if someone for some reason plugged in a transformation id here.

So perhaps we should filter for `type: texture` in the query as well

Reply:
Done! (It now queries only for textures and returns a 404 if a node isn't found.)